force-app/main/default/classes/ClassesObjectsMethods.cls

```diff
@@ -214,8 +290,15 @@ public with sharing class ClassesObjectsMethods {
     public static Integer calculateYearDifference(Date d1, Date d2) {
 
         // Subtract the difference in years
+        if (d1 == null || d2 == null) {
+            return null;
+        }
+
+        integer yearDiff = 0;
+
+        yearDiff = math.abs(d1.year() - d2.year());
 
-        return null; // Replace null with the variable you used to store the result
+        return yearDiff; // Replace null with the variable you used to store the result
     }
```

Comment:
No need to declare yearDiff = 0 on one line and give a new value on the next. 

If the values are not null you can return directly.